src/views/MoviesList/index.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { ReactElement, useEffect, useState } from 'react'
+import { NativeScrollEvent, ScrollView, View } from 'react-native'
+import type {
+  NavigationContainerRef,
+  RouteProp,
+} from '@react-navigation/native'
+
+import { API_VERSION, MOVIE_DB_API_KEY } from '@src/credentials'
+
+import Error from '@components/Error'
+import Image from '@components/Image'
+import MovieCard from '@components/MovieCard'
+import Layout from '@components/Layout'
+import Loader from '@components/Loader'
+import RatingUsCard from '@components/RatingUsCard'
+import Text from '@components/Text'
+
+import type { MovieDetails } from '@src/types'
+import { spaces } from '@src/styles/theme'
+import { gendersIcon } from '@src/views/Movies/components/GenderCard/constants'
+
+interface Props {
+  navigation: NavigationContainerRef
+  route: RouteProp<
+    {
+      params: {
+        movieType: 'popular' | 'upcoming' | 'genre' | 'search'
+        genreId?: number
+        genreName?: string
+        searchQuery?: string
+      }
+    },
+    'params'
+  >
+}
+
+const MoviesList = ({ navigation, route }: Props) => {
+  const [moviesList, setMoviesList] = useState<MovieDetails[]>([])
+  const [isDataFetching, setIsDataFetching] = useState<boolean>(true)
+  const [shouldDisplayError, setShouldDisplayError] = useState<boolean>(false)
+  const [isFetchingNewMovies, setIsFetchingNewMovies] = useState<boolean>(false)
+  const [page, setPage] = useState<number>(1)
+
+  let url = ''
+  let pageTitle: string | ReactElement = ''
+
+  switch (route.params.movieType) {
+    case 'popular':
+      pageTitle = 'Films populaire'
+      url = `https://api.themoviedb.org/${API_VERSION}/movie/popular?api_key=${MOVIE_DB_API_KEY}&language=fr`
+      break
+
+    case 'upcoming':
+      pageTitle = 'Prochaine sortie'
+      url = `https://api.themoviedb.org/${API_VERSION}/movie/upcoming?api_key=${MOVIE_DB_API_KEY}&language=fr`
+      break
+
+    case 'search':
+      if (!route.params.searchQuery) break
+      pageTitle = route.params.searchQuery
+      url = `https://api.themoviedb.org/${API_VERSION}/search/movie?api_key=${MOVIE_DB_API_KEY}&query=${route.params.searchQuery}&language=fr`
```

Comment:
&language=fr
to be reviewed when we will integrate the languages in the application

Reply:
Nothing change here ?